src/OpcUaStackCore/Core/Core.cpp

```diff
@@ -124,165 +124,81 @@ namespace OpcUaStackCore
-		eo.registerFactoryElement<ReadRawModifiedDetails>(OpcUaId_ReadRawModifiedDetails_Encoding_DefaultBinary);
-		eo.registerFactoryElement<ReadProcessedDetails>(OpcUaId_ReadProcessedDetails_Encoding_DefaultBinary);
-		eo.registerFactoryElement<ReadAtTimeDetails>(OpcUaId_ReadAtTimeDetails_Encoding_DefaultBinary);
-		eo.registerFactoryElement<HistoryData>(OpcUaId_HistoryData_Encoding_DefaultBinary);
-		eo.registerFactoryElement<HistoryModifiedData>(OpcUaId_HistoryModifiedData_Encoding_DefaultBinary);
-		eo.registerFactoryElement<ModificationInfo>(OpcUaId_ModificationInfo_Encoding_DefaultBinary);
-		eo.registerFactoryElement<HistoryEvent>(OpcUaId_HistoryEvent_Encoding_DefaultBinary);
-		eo.registerFactoryElement<UpdateDataDetails>(OpcUaId_UpdateDataDetails_Encoding_DefaultBinary);
-		eo.registerFactoryElement<HistoryUpdateDetails>(OpcUaId_HistoryUpdateDetails_Encoding_DefaultBinary);
-		eo.registerFactoryElement<UpdateStructureDataDetails>(OpcUaId_UpdateStructureDataDetails_Encoding_DefaultBinary);
-		eo.registerFactoryElement<UpdateEventDetails>(OpcUaId_UpdateEventDetails_Encoding_DefaultBinary);
-		eo.registerFactoryElement<DeleteRawModifiedDetails>(OpcUaId_DeleteRawModifiedDetails_Encoding_DefaultBinary);
-		eo.registerFactoryElement<DeleteAtTimeDetails>(OpcUaId_DeleteAtTimeDetails_Encoding_DefaultBinary);
-		eo.registerFactoryElement<DeleteEventDetails>(OpcUaId_DeleteEventDetails_Encoding_DefaultBinary);
-		eo.registerFactoryElement<ObjectAttributes>(OpcUaId_ObjectAttributes_Encoding_DefaultBinary);
-		eo.registerFactoryElement<NodeAttributes>(OpcUaId_NodeAttributes_Encoding_DefaultBinary);
-		eo.registerFactoryElement<VariableAttributes>(OpcUaId_VariableAttributes_Encoding_DefaultBinary);
-		eo.registerFactoryElement<VariableTypeAttributes>(OpcUaId_VariableTypeAttributes_Encoding_DefaultBinary);
-		eo.registerFactoryElement<MethodAttributes>(OpcUaId_MethodAttributes_Encoding_DefaultBinary);
-		eo.registerFactoryElement<ObjectTypeAttributes>(OpcUaId_ObjectTypeAttributes_Encoding_DefaultBinary);
-		eo.registerFactoryElement<DataTypeAttributes>(OpcUaId_DataTypeAttributes_Encoding_DefaultBinary);
-		eo.registerFactoryElement<ReferenceTypeAttributes>(OpcUaId_ReferenceTypeAttributes_Encoding_DefaultBinary);
-		eo.registerFactoryElement<ViewAttributes>(OpcUaId_ViewAttributes_Encoding_DefaultBinary);
-		eo.registerFactoryElement<AggregateFilter>(OpcUaId_AggregateFilter_Encoding_DefaultBinary);
-		eo.registerFactoryElement<Range>(OpcUaId_Range_Encoding_DefaultBinary);
-		eo.registerFactoryElement<EUInformation>(OpcUaId_EUInformation_Encoding_DefaultBinary);
-
-		// xml
-		eo.registerFactoryElement<Argument>(OpcUaId_Argument_Encoding_DefaultXml);
-		eo.registerFactoryElement<BuildInfo>(OpcUaId_BuildInfo_Encoding_DefaultXml);
-		eo.registerFactoryElement<EnumValueType>(OpcUaId_EnumValueType_Encoding_DefaultXml);
-
-		// json
+		eo.registerFactoryObject<LiteralOperand>();
```

Comment:
I offer to check the result of registration like this:

```cpp
rc &= eo.registerFactoryObject<Range>();
```

with an exception or `assert(rc)` in the end because the consequences of the wrong registration appear in different places of the project.

Reply:
Will be done with bug https://github.com/ASNeG/OpcUaStack/issues/216